### PR TITLE
Core/Spells: players shouldn't force their orientation (serverside) on each periodictick when channeling

### LINKED ISSUE #26095: Channeling spells allow all other spells to be casted on a back-facing target
**Description:**

After you change your character orientation when you channeling any spell , you can cast any other spell with back facing target.

**Expected behaviour:**

You shoudn't be able to cast all spells to a back facing target.

**Steps to reproduce the problem:**

1. Cast any channeling spell , for example Drain Soul .
2. Change your character position with left click + left / right keyboard arrows (you need to stay in place).
3. Now you are able to cast any other spell to back facing target .

**Branch(es):**

3.3.5

**TC rev. hash/commit:** f6e52f6b3df9

TrinityCore rev. f6e52f6b3df9 2021-02-14 14:38:06 +0100 (3.3.5 branch) (Win64, Release, Static)

**Operating system:** 

Windows 10 x64

https://user-images.githubusercontent.com/24683355/107940911-a25ee080-6f91-11eb-8e0a-3c3a5eada151.mp4

### PR DESCRIPTION
**Pull request by Jildor**
**Changes proposed:**

-  exclude players form force their orientation (serverside) on each periodictick when channeling because can turn during channeling and shouldn't desync orientation client/server

**Target branch(es):** 3.3.5/master

- [X] 3.3.5
- [ ] master

**Issues addressed:**

Closes #26095

**Tests performed:**

tested in-game

### CODE REVIEW

File: src/server/game/Spells/Auras/SpellAuraEffects.cpp [hunk 1]
```diff
@@ -869,7 +869,8 @@ void AuraEffect::PeriodicTick(AuraApplication* aurApp, Unit* caster) const
     Unit* target = aurApp->GetTarget();
 
     // Update serverside orientation of tracking channeled auras on periodic update ticks
-    if (caster && m_spellInfo->IsChanneled() && m_spellInfo->HasAttribute(SPELL_ATTR1_CHANNEL_TRACK_TARGET))
+    // exclude players because can turn during channeling and shouldn't desync orientation client/server
+    if (caster && caster->GetTypeId() != TYPEID_PLAYER && m_spellInfo->IsChanneled() && m_spellInfo->HasAttribute(SPELL_ATTR1_CHANNEL_TRACK_TARGET))
```
Comment (Ovahlord) [inline src/server/game/Spells/Auras/SpellAuraEffects.cpp:873]: !caster->IsPlayer()   we're not in 2008 anymore

Comment (Jildor) [reply src/server/game/Spells/Auras/SpellAuraEffects.cpp]: thx ;)

**Comment by CraftedRO** [comment]
I tested ingame this PR and indeed it fix the casting of other spells with back facing but you can still back-turn when channeling without intreruptions.

**Comment by Jildor** [comment]
That is how should ir work

**Comment by CraftedRO** [comment]
im not so sure about that but for a moment is fine , if I channel as priest for example mind fly on a rogue and by mystake I turned back to him , he can back stab or whathever want me when the behavior at those channeling things is to be sure you are facing the direction of enemy if I remember right and somehow is kinda logic.

**Comment by jackpoz** [comment]
should the client side orientation be changed instead when channeling ?

**Comment by Jildor** [comment]
No, client side orientation is updated when cast the spell, but after that seems shouldn't change for player caster

**Comment by jackpoz** [comment]
and is that the retail behavior too ?

**Comment by Jildor** [comment]
https://www.youtube.com/watch?v=dVu80XgRZB4 <-- 4:48 cast Drain Life on feared player and I don't see how change player caster orientation, 5:40 cast Drain Soul on bladestorm warrior and warrior stands on his back

**Comment by jackpoz** [comment]
let's merge this then and change it in the future if anyone has a different opinion

**Comment by jackpoz** [comment]
Thanks for the PR :)

**Pull request merged**